include/realtime_tools/realtime_barrier.hpp

```diff
@@ -0,0 +1,468 @@
+  /// Copy the current data into dest. No swap is performed and the new_data_available flag is
+  /// unaffected.
+  bool current(T & dest)
+  {
+    typename MemoryBarrierType::template DirectAccess<non_realtime, locking_strategy> direct(*mem_);
+    if (direct) {
+      dest = *direct;
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  /// @brief Read data out of the realtime thread
+  /// Swap RT buffer for non-RT. Copy the new data into val and reset the new_data_available flag.
+  // todo: since this read also swaps, should it be called read_and_swap() or pop() or pull()
+  bool pull(T & dest)
+  {
+    typename MemoryBarrierType::template DirectAccess<non_realtime, locking_strategy> direct(*mem_);
+    if (direct && mem_->new_data_available()) {
+      direct.swap();
+      direct.new_data_available(false);
+      dest = *direct;
+      return true;
+    } else {
+      return false;     // failed to read
+    }
+  }
+
+private:
+  MemoryBarrierType * mem_;
+  bool owns_mem;
+};
+
+
+/// @brief Create a barrier for writing data to a realtime thread
+/// WriteBarrier implements writing data to a realtime thread from a non-realtime thread using a
+/// MemoryBarrier. For example, this is used to transfer commands to a hardware interface. The
+/// default constructor will create a new memory barrier for use. There is an alternate
+/// constructor if you want to create your own MemoryBarrier explicitly.
+template<class T, class locking_strategy = wait_lock>
+class WriteBarrier
+{
+public:
+  using MemoryBarrierType = MemoryBarrier<T>;
+
+  WriteBarrier() noexcept
+  : mem_(new MemoryBarrierType())
+  {
+  }
+
+  explicit WriteBarrier(MemoryBarrierType & mem_barrier) noexcept
+  : mem_(&mem_barrier)
+  {
+  }
+
+  /// @brief Access the underlying MemoryBarrier object
+  inline MemoryBarrierType & memory()
+  {return *mem_;}
+
+  /// @brief Get current value from non-realtime side without affecting the barrier.
+  /// Copy the current data into dest. No swap is performed and the new_data_available flag
+  /// is unaffected.
+  bool current(T & dest)
+  {
+    typename MemoryBarrierType::template DirectAccess<non_realtime, locking_strategy> direct(*mem_);
+    if (direct) {
+      dest = *direct;
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  /// @brief Write data into the realtime thread
+  /// Copy the new data from val into non-RT buffer then swap RT buffer for non-RT and set the
+  // new_data_available flag.
+  bool push(const T & data)
+  {
+    typename MemoryBarrierType::template DirectAccess<non_realtime, locking_strategy> direct(*mem_);
+    if (direct) {
+      *direct = data;
+      direct.swap();
+      direct.new_data_available(true);
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+private:
+  MemoryBarrierType * mem_;
+};
+
+}  // namespace realtime_tools
+
+
+#endif    // REALTIME_TOOLS__REALTIME_BARRIER_HPP_
```

Comment:
Should be 2 spaces after end of line, before comment. I think this fix is needed in a handful of places